custom_components/tuya_local/devices/hlg_infinity_alarm.yaml

```diff
@@ -0,0 +1,332 @@
+          - dps_val: PANICO_ON
+            value: PANIC_ON
+          - dps_val: PANICO_OFF
+            value: PANIC_OFF
+          - dps_val: SIRENE_ON
+            value: SIREN_ON
+          - dps_val: SIRENE_OFF
+            value: SIREN_OFF
+          - dps_val: REINICIADO
+            value: RESTARTED
+  - entity: switch
+    name: Beep Siren When Sector On/Off
+    category: config
+    dps:
+      - id: 110
+        type: boolean
+        name: switch
+  - entity: select
+    name: Learn Remote Control
+    icon: "mdi:remote"
+    category: config
+    dps:
+      - id: 111
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "S1_S2_S3_S4"
+            value: "All Sectors"
+          - dps_val: "S1"
+            value: "Sector 1"
+          - dps_val: "S2"
+            value: "Sector 2"
+          - dps_val: "S3"
+            value: "Sector 3"
+          - dps_val: "S4"
+            value: "Sector 4"
+          - dps_val: "S1_S2"
+            value: "Sectors: 1 and 2"
+          - dps_val: "S3_S4"
+            value: "Sectors: 3 and 4"
+          - dps_val: "S1_S3"
+            value: "Sectors: 1 and 3"
+          - dps_val: "S1_S4"
+            value: "Sectors: 1 and 4"
+          - dps_val: "S2_S3"
+            value: "Sectors: 2 and 3"
+          - dps_val: "S2_S4"
+            value: "Sectors: 2 and 4"
+          - dps_val: "S1_S2_S3"
+            value: "Sectors: 1, 2 and 3"
+          - dps_val: "S2_S3_S4"
+            value: "Sectors: 2, 3 and 4"
+          - dps_val: "S1_S3_S4"
+            value: "Sectors: 1, 3 and 4"
+          - dps_val: "S1_S2_S4"
+            value: "Sectors: 1, 2 and 4"
+          - dps_val: "PANICO"
+            value: "Panic"
+          - dps_val: "CANCEL"
+            value: "Cancel"
+  - entity: select
+    name: Learn Sensor
+    icon: "mdi:motion-sensor"
+    category: config
+    dps:
+      - id: 112
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: "S1"
+            value: "Sector 1"
+          - dps_val: "S2"
+            value: "Sector 2"
+          - dps_val: "S3"
+            value: "Sector 3"
+          - dps_val: "S4"
+            value: "Sector 4"
+          - dps_val: "CANCEL"
+            value: "Cancel"
+  - entity: sensor
+    name: Number of Buttons
+    icon: "mdi:remote"
+    category: diagnostic
+    dps:
+      - id: 113
+        type: integer
+        name: sensor
+        unit: "buttons"
+  - entity: sensor
+    name: Number of Sensors
+    icon: "mdi:motion-sensor"
+    category: diagnostic
+    dps:
+      - id: 114
+        type: integer
+        name: sensor
+        optional: true
+        unit: "sensors"
+  - entity: binary_sensor
+    name: Power Line
+    icon: "mdi:power-plug"
+    category: diagnostic
+    dps:
+      - id: 115
+        type: boolean
+        name: sensor
+        mapping:
+          - dps_val: true
+            value: "Power Line"
```

Comment:
Values should be `true` and `false` for binary sensors (no mapping required) and a `class: plug` would be appropriate here. The class defines an icon, and translated name, which might be better than explicitly hardcoding them.

Reply:
Fixed!